types/index.d.ts

```diff
@@ -143,17 +143,44 @@ declare namespace Realm {
     }
 
     enum ClientResetMode {
-        Manual = "manual",
-        DiscardLocal = "discardLocal",
+        Manual = 'manual',
+        DiscardLocal = 'discardLocal',
+        Recover = 'recover',
+        RecoverOrDiscard = 'recoverOrDiscard'
     }
 
+    enum ClientResetDidRecover {
+      Yes = 'yes',
+      No = 'no'
+    }
+
+    type ClientResetManualCallback = (session: Realm.App.Sync.Session, path: string) => void;
     type ClientResetBeforeCallback = (localRealm: Realm) => void;
     type ClientResetAfterCallback = (localRealm: Realm, remoteRealm: Realm) => void;
-    interface ClientResetConfiguration<ClientResetModeT = ClientResetMode> {
-        mode: ClientResetModeT;
-        clientResetBefore?: ClientResetBeforeCallback;
-        clientResetAfter?: ClientResetAfterCallback;
+    type ClientResetAfterRecoveryCallback = (localRealm: Realm, remoteRealm: Realm, didRecover: ClientResetDidRecover) => void;
```

Comment:
Could we merge the `ClientResetAfterCallback` and `ClientResetAfterRecoveryCallback` if we provided `false` as default value for `didRecover`?

Reply:
`didRecover` is something the sync client sets. Moreover, I want to express that `didRecover` do not make sense in all cases.

Reply:
I think it should simply call this with false if it didn't happen. Users providing this callback can simply choose not to read that third argument.